codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpBoundProtocolGenerator.kt

```diff
@@ -1005,85 +989,69 @@ private class ServerHttpBoundProtocolTraitImplGenerator(
                 *codegenScope,
                 "O" to output,
             ) {
-                // `<_>::from()` is necessary to convert the `&str` into:
-                //     * the Rust enum in case the `string` shape has the `enum` trait; or
-                //     * `String` in case it doesn't.
-                when (symbol.rustType()) {
-                    RustType.String ->
```

Comment:
This is the branch the `From<std::convert::Infallible> for RequestRejection` removes. Worth it?

Reply:
There is another branch which I noted for removal - https://github.com/awslabs/smithy-rs/pull/1417/files#diff-da9c47cae3f2f75fc01b3625ee430f56f2f495870a77302c3c148e98f7341bd4R860. I opted out of removing it because I was less sure.

Reply:
I've taken this second branch out in https://github.com/awslabs/smithy-rs/pull/1417/commits/1477e678baa73c52510c12d9ccae639e650d2179